src/effects/VST3/VST3Effect.cpp

```diff
@@ -350,58 +350,48 @@ bool VST3Effect::SupportsAutomation() const
 bool VST3Effect::SaveSettings(
    const EffectSettings &, CommandParameters & parms) const
 {
-   if(mEditController == nullptr)
-      return false;
-
-   using namespace Steinberg;
-   
-   for(int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
-   {
-      Vst::ParameterInfo parameterInfo { };
-      if(mEditController->getParameterInfo(i, parameterInfo) == kResultOk)
+   return ForEachParameter
+   (
+      [&](const ParameterInfo& pi)
       {
-         if(parameterInfo.flags & Vst::ParameterInfo::kCanAutomate)
-         {
-            parms.Write(
-               VST3Utils::MakeAutomationParameterKey(parameterInfo),
-               mEditController->getParamNormalized(parameterInfo.id)
-            );
-         }
-      }
-   }
+         parms.Write(
+            VST3Utils::MakeAutomationParameterKey(pi),
+            mEditController->getParamNormalized(pi.id)
+         );
 
-   return true;
+         return true;
+      }
+   );
 }
 
 bool VST3Effect::LoadSettings(
    const CommandParameters & parms, EffectSettings &settings) const
 {
    using namespace Steinberg;
 
-   if(mComponentHandler == nullptr)
+   if (mComponentHandler == nullptr)
       return false;
-   
-   long index { };
-   wxString key;
-   if(parms.GetFirstEntry(key, index))
-   {
-      do
+
+   ForEachParameter
+   (
+      [&](const ParameterInfo& pi)
       {
-         Steinberg::Vst::ParamID id;
-         Vst::ParamValue value;
-         if(VST3Utils::ParseAutomationParameterKey(key, id) && parms.Read(key, &value))
+         const auto id = pi.id;
+         const auto value = pi.defaultNormalizedValue;
+
+         if (mComponentHandler->beginEdit(id) == kResultOk)
          {
-            if(mComponentHandler->beginEdit(id) == kResultOk)
-            {
-               auto cleanup = finally([&]{
-                  mComponentHandler->endEdit(id);
-               });
-               mComponentHandler->performEdit(id, value);
-            }
-            mEditController->setParamNormalized(id, value);
+            auto cleanup = finally([&] {
+               mComponentHandler->endEdit(id);
+            });
+            mComponentHandler->performEdit(id, value);
          }
-      } while(parms.GetNextEntry(key, index));
-   }
+         mEditController->setParamNormalized(id, value);
```

Comment:
The second step of the transformation of this function is meant to eliminate all the uses of the handles, but you have not done that yet.

When you define `StoreSettings`, it introduces a duplication of the use of the edit controller -- but don't eliminate this call yet.

When you define `TransferDataToWindow`, you eliminate the duplicate lines that use the component handler, which move to that function.  And because `TransferDataToWindow` inserts a call to `StoreSettings`, then at this step you can eliminate the duplicate use of the edit controller here.  You should be able to verify with testing of an effect, that there is no loss of response of the effect to changes of the dials.

So then perhaps all those prior steps will leave nothing to do in `LoadSettings` at this late step.